indexers/src/main/java/org/mousephenotype/cda/indexers/GenotypePhenotypeIndexer.java

```diff
@@ -328,32 +331,36 @@ public int populateGenotypePhenotypeSolrCore(RunStatus runStatus) throws SQLExce
                     doc.setMpTermId(mpId);
                     doc.setMpTermName(r.getString("ontology_term_name"));
 
-                    OntologyTermHelperMp mpTerm = new OntologyTermHelperMp(mpOntologyService, mpId);
                     // mp-anatomy mappings (all MA at the moment)
-                    if (mpTerm.getAnatomyMappings() != null){
-                    	List<String> anatomyIds = mpTerm.getAnatomyMappings();
+                    if (mpOntologyService.getAnatomyMappings(mpId) != null){
+                    	List<String> anatomyIds = mpOntologyService.getAnatomyMappings(mpId);
                     	for (String id: anatomyIds){
-                    		OntologyTermHelperMa maTerm = new OntologyTermHelperMa(maOntologyService, id);
+                    		OntologyDAO currentOntologyService = null;
+                    		if (id.startsWith("EMAPA")){
+                    			currentOntologyService = emapaOntologyService;
+                    		} else  if (id.startsWith("MA")){
+                    			currentOntologyService = maOntologyService;
+                    		}
                     		doc.addAnatomyTermId(id);
-                    		doc.addAnatomyTermName(maTerm.getOntologyTerm().getName());
-                    		OntologyDetail maAncestors = maTerm.getIntermediates();
+                    		doc.addAnatomyTermName(currentOntologyService.getTerm(id).getName());
+                    		OntologyDetail maAncestors = currentOntologyService.getIntermediatesDetail(id);
```

Comment:
Are ancestors and intermediates the same thing here?  Should this be a call to getAncestors?